Mathlib/Algebra/BigOperators/Order.lean

```diff
@@ -845,7 +845,7 @@ TC inference. -/
 def evalFinsetSum : PositivityExt where eval {u α} zα pα e := do
   match e with
   | ~q(@Finset.sum _ $ι $instα $s $f) =>
-    let (lhs, _, (rhs : Q($α))) ← lambdaMetaTelescope f
+    let (lhs, _, (rhs : Q($α))) ← lambdaMetaTelescope (if f.isLambda then f else q(fun x => $f x))
```

Comment:
This looks risky if you have `f := fun {ohno} (x) => ohno`; though I suspect this is not the right approach anyway.

Reply:
Sorry I'm not sure I understand your example can you elaborate on that the issue would be?

Reply:
`f := fun {ohno} (x) => ohno` can't happen as an argument to `Finset.sum`?

Reply:
> can you elaborate on that the issue would be?

`$f x` assumes that the argument to `f` is explicit



> `f := fun {ohno} (x) => ohno` can't happen as an argument to `Finset.sum`?

Isn't `Finset.sum univ fun {ohno : ℕ} (x : ℕ) => ohno` legal?

Reply:
Oh sure, but in your example `(fun {ohno : ℕ} (x : ℕ) => ohno) 1` is just id in function space, which seems like it would work fine.
If you can come up with a realistic mw `example` I'll try and fix, but I can't really think of any myself.

Reply:
I don't know what you mean by it being `id`; it's not a function at all  `(fun {ohno : ℕ} (x : ℕ) => ohno) 1 : ℕ`, and it reduces to `?ohno : ℕ` with an unresolved metavariable

Reply:
`Finset.sum univ (id fun {n : ℕ} => n)` feels like a more realistic test

Reply:
I see, I wasn't aware it would eagerly apply implicits in this way

Reply:
I mean like realistic in the sense of something someone will be applying positivity too. Its nice to be feature complete but I'm not sure how much time I want to spend fixing things that will never come up in practice, at the moment its still not clear to me whether the issue you're pointing out is a real issue for this tactic or not

Reply:
My example was more to demonstrate that this approach seems somewhat fishy; it's fine for positivity to not support these cases, but it should never produce invalid proof terms.